### PR TITLE
[Merged by Bors] - refactor: use framework LDAP functions

### PR DESCRIPTION
**Pull request by fhennig**
# Description

*Please add a description here. This will become the commit message of the merge request later.*

## Review Checklist

- [ ] Code contains useful comments
- [ ] CRD change approved (or not applicable)
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Comment by fhennig** [comment]
Test started: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/superset-operator-it-custom/23/

File: rust/operator-binary/src/superset_controller.rs [hunk 1]
```diff
@@ -223,6 +216,8 @@ pub async fn reconcile_superset(superset: Arc<SupersetCluster>, ctx: Arc<Ctx>) -
         return Ok(Action::await_change());
     }
 
+    // Since rust 1.64 the '&*' became a clippy warning. It is needed though to dereference the Arc here
+    #[allow(clippy::explicit_auto_deref)]
```
Comment (maltesander) [inline rust/operator-binary/src/superset_controller.rs:220]: Try `superset.as_ref()` and remove the comment and clippy allow
```suggestion
```

**Review by maltesander — CHANGES_REQUESTED** [review]
LGTM! Nice code reduction!

File: rust/operator-binary/src/superset_controller.rs (hunk 1 above)
Comment (fhennig) [reply rust/operator-binary/src/superset_controller.rs]: good suggestion, thanks! fixed: [08ff36f](https://github.com/stackabletech/superset-operator/pull/311/commits/08ff36fbc9ad0af03fdb27d48e12ef3a39ed58f6)

**Review by maltesander — APPROVED** [review]
LGTM. Lets wait for a templating PR to bump the rust version.

**Comment by fhennig** [comment]
bors merge

**Pull request closed without merging**